core/include/utils/enumerate.hpp

```diff
@@ -19,11 +20,11 @@ namespace detray {
  * @tparam volume_type the type of volume from which to get the range in the
  *         container
  */
-template <typename container_type,
-          typename container_type_iter =
-              decltype(std::begin(std::declval<container_type>())),
-          typename = decltype(std::end(std::declval<container_type>()))>
+template <typename container_type>
```

Comment:
I think the other structs doesn't show error because they are not used in cuda currently.
And I am almost sure that it is buggy behavior of cuda 11.3 - didn't see the error in cuda 11.2